src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -571,11 +570,13 @@ protected internal override Task OnCloseAsync(TimeSpan timeout)
         protected override void OnClosed()
         {
             base.OnClosed();
-            var httpClientToDispose = _httpClient;
-            if (httpClientToDispose != null)
+            if (_httpClientCache != null && !_httpClientCache.IsDisposed)
             {
-                _httpClient = null;
-                httpClientToDispose.Dispose();
+                lock (_httpClientCache)
```

Comment:
Also there appears to be a get here before _httpClientCache is even created and then create _httpClientCache on another thread.
The window for this is small though and as far as I can see the worst case is we'll end up with a closed factory holding a reference to MRU cache (which is not worse than before and it would require another lock here which I am ware about).